packages/client/lib/execution/vmexecution.ts

```diff
@@ -302,82 +302,105 @@ export class VMExecution extends Execution {
       headBlock = undefined
       parentState = undefined
       errorBlock = undefined
-      this.vmPromise = blockchain.iterator(
-        'vm',
-        async (block: Block, reorg: boolean) => {
-          if (errorBlock !== undefined) return
-          // determine starting state for block run
-          // if we are just starting or if a chain reorg has happened
-          if (headBlock === undefined || reorg) {
-            const headBlock = await blockchain.getBlock(block.header.parentHash)
-            parentState = headBlock.header.stateRoot
-
-            if (reorg) {
-              this.config.logger.info(
-                `Chain reorg happened, set new head to block number=${headBlock.header.number}, clearing state cache for VM execution.`
-              )
-            }
-          }
-          // run block, update head if valid
-          try {
-            const { number, timestamp } = block.header
-            if (typeof blockchain.getTotalDifficulty !== 'function') {
-              throw new Error(
-                'cannot get iterator head: blockchain has no getTotalDifficulty function'
-              )
+      this.vmPromise = blockchain
```

Comment:
again most of diff here is because of formatting changed because of introduction to the `catch` to the iterator promise below